gravitee-apim-gateway/gravitee-apim-gateway-flow/src/main/java/io/gravitee/gateway/reactive/flow/AbstractFlowResolver.java

```diff
@@ -35,4 +45,49 @@ protected AbstractFlowResolver(ConditionFilter<Flow> filter) {
     public Flowable<Flow> resolve(GenericExecutionContext ctx) {
         return provideFlows(ctx).flatMapMaybe(flow -> filter.filter(ctx, flow));
     }
+
+    protected void addContextRequestPathParameters(GenericExecutionContext context, List<Flow> flows) {
+        flows.stream().forEach(flow -> addContextRequestPathParameters(context, flow));
+    }
+
+    protected void addContextRequestPathParameters(GenericExecutionContext context, Flow flow) {
+        if (flow.getPath() == null || flow.getPath().isEmpty()) {
+            return;
+        }
+
+        Map<Integer, String> parameters = new HashMap();
+        String[] branches = URL_PATH_SEPARATOR_PATTERN.split(flow.getPath());
+
+        for (int position = 0; position < branches.length; position++) {
+            final String branch = branches[position];
+            if (branch.startsWith(PATH_PARAM_PREFIX)) {
+                parameters.put(position, branch.substring(PATH_PARAM_PREFIX.length()));
+            }
+        }
+
+        if (parameters.isEmpty()) {
+            return;
+        }
+
+        int count = 0;
+        int off = 0;
+        int next;
+
+        final String pathInfo = context.request().pathInfo();
+        final Iterator<Integer> iterator = parameters.keySet().iterator();
+
+        Integer currentPosition = iterator.next();
+        while ((next = pathInfo.indexOf(URL_PATH_SEPARATOR, off)) != -1) {
+            if (count == currentPosition) {
+                context.request().pathParameters().add(parameters.get(currentPosition), pathInfo.substring(off, next));
+                if (iterator.hasNext()) {
+                    currentPosition = iterator.next();
+                }
+            }
+            off = next + 1;
+            count++;
+        }
+
+        context.request().pathParameters().add(parameters.get(currentPosition), pathInfo.substring(off));
+    }
```

Comment:
This is probably something we could try to mutualize for both v2 and v4 to avoid duplicating this algorithm and ease the maintenance. wdyt?